pkg/services/rpcsrv/server_test.go

```diff
@@ -2275,96 +2275,223 @@ func TestSubmitOracle(t *testing.T) {
-	t.Run("invalid request bytes", runCase(t, true, neorpc.InvalidParamsCode, `"not-a-request"`))
-	t.Run("invalid request", func(t *testing.T) {
-		mainTx := &transaction.Transaction{
-			Attributes:      []transaction.Attribute{{Type: transaction.NotaryAssistedT, Value: &transaction.NotaryAssisted{NKeys: 1}}},
-			Script:          []byte{byte(opcode.RET)},
-			ValidUntilBlock: 123,
-			Signers:         []transaction.Signer{{Account: util.Uint160{1, 5, 9}}},
-			Scripts: []transaction.Witness{{
-				InvocationScript:   []byte{1, 4, 7},
-				VerificationScript: []byte{3, 6, 9},
-			}},
+
+	t.Run("getrawnotarypool", func(t *testing.T) {
+		t.Run("empty pool", func(t *testing.T) {
+			body := doRPCCallOverHTTP(rpcPool, httpSrv.URL, t)
+			res := checkErrGetResult(t, body, false, 0)
+			actual := new(result.RawNotaryPool)
+			require.NoError(t, json.Unmarshal(res, actual))
+			require.Equal(t, 0, len(actual.Hashes))
+		})
+
+		sender := testchain.PrivateKeyByID(0) // owner of the deposit in testchain
+		notaryRequest1 = createValidNotaryRequest(chain, sender, 1, 2_0000_0000, &transaction.Transaction{})
+		nrBytes, err := notaryRequest1.Bytes()
+		require.NoError(t, err)
+		str := fmt.Sprintf(`"%s"`, base64.StdEncoding.EncodeToString(nrBytes))
+		submitNotaryRequest(t, false, 0, str)(t)
+
+		t.Run("nonempty pool", func(t *testing.T) {
+			//get notary pool & check tx hashes
+			body := doRPCCallOverHTTP(rpcPool, httpSrv.URL, t)
+			res := checkErrGetResult(t, body, false, 0)
+			actual := new(result.RawNotaryPool)
+			require.NoError(t, json.Unmarshal(res, actual))
+			require.Equal(t, 1, len(actual.Hashes))
+			for actMain, actFallbacks := range actual.Hashes {
+				require.Equal(t, notaryRequest1.MainTransaction.Hash(), actMain)
+				require.Equal(t, 1, len(actFallbacks))
+				require.Equal(t, notaryRequest1.FallbackTransaction.Hash(), actFallbacks[0])
+			}
+		})
+
+		notaryRequest2 = createValidNotaryRequest(chain, sender, 2, 3_0000_0000, notaryRequest1.MainTransaction)
+		nrBytes2, err := notaryRequest2.Bytes()
+		require.NoError(t, err)
+		str2 := fmt.Sprintf(`"%s"`, base64.StdEncoding.EncodeToString(nrBytes2))
+		submitNotaryRequest(t, false, 0, str2)(t)
+
+		t.Run("pool with 2", func(t *testing.T) {
+			//get notary pool & check tx hashes
+			body := doRPCCallOverHTTP(rpcPool, httpSrv.URL, t)
+			res := checkErrGetResult(t, body, false, 0)
+			actual := new(result.RawNotaryPool)
+			require.NoError(t, json.Unmarshal(res, actual))
+			require.Equal(t, 1, len(actual.Hashes))
+			for actMain, actFallbacks := range actual.Hashes {
+				require.Equal(t, notaryRequest1.MainTransaction.Hash(), actMain)
+				require.Equal(t, 2, len(actFallbacks))
+				// The second notary request has higher priority, so it's first in the slice.
+				require.Equal(t, notaryRequest1.FallbackTransaction.Hash(), actFallbacks[1])
+				require.Equal(t, notaryRequest2.FallbackTransaction.Hash(), actFallbacks[0])
```

Comment:
s/`The second notary request`/`The second fallback transaction`

Reply:
Fixed.